crates/ruff_linter/src/settings/mod.rs

```diff
@@ -38,13 +37,133 @@ pub mod flags;
+/// ```
+/// use std::fmt;
+/// use ruff_linter::display_settings;
+/// #[derive(Default)]
+/// struct Settings {
+///     option_a: bool,
+///     sub_settings: SubSettings,
+///     option_b: String,
+/// }
+///
+/// struct SubSettings {
+///     name: String
+/// }
+///
+/// impl Default for SubSettings {
+///     fn default() -> Self {
+///         Self { name: "Default Name".into() }
+///     }
+///
+/// }
+///
+/// impl fmt::Display for SubSettings {
+///     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+///         display_settings! {
+///             formatter = f,
+///             namespace = "sub_settings",
+///             fields = [
+///                 self.name | quoted
+///             ]
+///         }
+///         Ok(())
+///     }
+///
+/// }
+///
+/// impl fmt::Display for Settings {
+///     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+///         display_settings! {
+///             formatter = f,
+///             fields = [
+///                 self.option_a,
+///                 self.sub_settings | nested,
+///                 self.option_b | quoted,
+///             ]
+///         }
+///         Ok(())
+///     }
+///
+/// }
+///
+/// const EXPECTED_OUTPUT: &str = r#"option_a = false
+/// sub_settings.name = "Default Name"
+/// option_b = ""
+/// "#;
+///
+/// fn main() {
+///     let settings = Settings::default();
+///     assert_eq!(format!("{settings}"), EXPECTED_OUTPUT);
+/// }
+/// ```
+#[macro_export]
+macro_rules! display_settings {
+    (formatter = $fmt:ident, namespace = $namespace:literal, fields = [$($settings:ident.$field:ident $(| $modifier:tt)?),* $(,)?]) => {
+        {
+            const _PREFIX: &str = concat!($namespace, ".");
+            $(
+                display_settings!(@field $fmt, _PREFIX, $settings.$field $(| $modifier)?);
+            )*
+        }
+    };
+    (formatter = $fmt:ident, fields = [$($settings:ident.$field:ident $(| $modifier:tt)?),* $(,)?]) => {
+        {
+            const _PREFIX: &str = "";
+            $(
+                display_settings!(@field $fmt, _PREFIX, $settings.$field $(| $modifier)?);
+            )*
+        }
+    };
+    (@field $fmt:ident, $prefix:ident, $settings:ident.$field:ident | debug) => {
+        writeln!($fmt, "{}{} = {:?}", $prefix, stringify!($field), $settings.$field)?;
+    };
+    (@field $fmt:ident, $prefix:ident, $settings:ident.$field:ident | quoted) => {
+        writeln!($fmt, "{}{} = \"{}\"", $prefix, stringify!($field), $settings.$field)?;
+    };
+    (@field $fmt:ident, $prefix:ident, $settings:ident.$field:ident | nested) => {
+        write!($fmt, "{}", $settings.$field)?;
+    };
+    (@field $fmt:ident, $prefix:ident, $settings:ident.$field:ident | optional) => {
+        {
+            write!($fmt, "{}{} = ", $prefix, stringify!($field))?;
+            match &$settings.$field {
+                Some(value) => writeln!($fmt, "{}", value)?,
+                None        => writeln!($fmt, "nil")?
```

Comment:
I'd go for `none` or `None` here, it's what rust (`Option::None`) and python (`None`) use.

```suggestion
                None        => writeln!($fmt, "none")?
```